Shear/InMemoryHistoryStorage.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  InMemoryHistoryStorage.swift
+//  Shear
+//
+//  Created by Andreas Kompanez on 23.02.20.
+//  Copyright © 2020 Andreas Kompanez. All rights reserved.
+//
+
+import Foundation
+
+class InMemoryHistoryStorage: HistoryStorageType {
+    private var storage: [HistoryItem] = []
+    
+    private var set: Set<HistoryItem> = []
+    
+    var count: Int {
+        return storage.count
+    }
+    
+    func store(item: HistoryItem, handler: (HistoryStorageStoreError?, HistoryItem) -> Void) {
+        if contains(item: item) {
+            handler(.duplicate, item)
+            return
+        }
+        set.insert(item)
+        storage.append(item)
+        handler(nil, item)
+    }
+    
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/kmpnz/Shear/blob/895e50bbb85826ea77037e4a1c1e5468d960c4fc/Shear/InMemoryHistoryStorage.swift#L29"></span>Lines should not have trailing whitespace.<br />
<code>trailing_whitespace</code> <code>InMemoryHistoryStorage.swift:29</code></del></td>
    </tr>
  </tbody>
</table>